docs/src/openapi-v1.yaml

```diff
@@ -643,6 +644,62 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /contracts/ink/{address}/query:
+    post:
+      tags:
+      - contracts
+      summary: Query an !Ink contract with a given message (method).
+      description: Will return a valid or invalid result. 
+      operationId: callContractQuery
+      requestBody:
+        $ref: '#/components/requestBodies/ContractMetadata'
+      parameters:
+      - name: address
+        in: path
+        description: SS58 or Hex address of the account associated with the contract.
+        required: true
+        schema:
+          pattern: a-km-zA-HJ-NP-Z1-9{8,64}
+          type: string
+      - name: method
+        in: query
+        description: The message or method used to query.
+        required: false
+        schema:
+          type: string
+          default: 'get'
+      - name: gasLimit
+        in: query
+        description: The gas limit to be used as an option for the queried message.
+        required: false
+        schema:
+          default: -1
+          type: number
+      - name: storageDepositLimit
+        in: query
+        description: The storage deposit limit to be used as an option for the queried message.
+        required: false
+        schema:
+          deafult: null
+          type: number
```

Comment:
Dumb question; what effect do these have (maybe I'll see if I look lower!)?

Does a contract have gas which is used even when something just queries it?